denoise.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+import numpy as np
+import wave
+import math
+
+def nextpow2(n):
+    return np.ceil(np.log2(np.abs(n))).astype('long')
+
+
+
+f = wave.open("denoise/output.wav")

```

Comment:
I think we need to further break down this script into discrete functions as there is no way at the moment to vary the inputs and outputs of the script without changing the code every single time. I propose you divide this code into an input function that takes a `str` parameter for the audio file path.

Reply:
I update the redirection function and encapsulate the main denoise function which accept a wav file.

Reply:
If I understand your words in a wrong way please let my know, I will fix it again.